sql/planbuilder/proc.go

```diff
@@ -119,15 +119,11 @@ func (p *procCtx) HasCursor(name string) bool {
 
 func (p *procCtx) AddHandler(h *plan.DeclareHandler) {
 	p.NewState(dsHandler)
-	if p.handler != nil {
-		err := sql.ErrDeclareHandlerDuplicate.New()
-		p.s.b.handleErr(err)
-	}
-	p.handler = h
+	p.handlers = append(p.handlers, h)
 }
 
 func (p *procCtx) HasHandler(name string) bool {
-	return p.handler != nil
+	return p.handlers != nil
```

Comment:
Yeah, tbh i don't know what `name` means here so I don't really know what to check. Maybe it's specifically for named signal handlers? But we don't support those. It could be by sqlstate also?